internal/substatoptimizer/regex.go

```diff
@@ -0,0 +1,67 @@
+package substatoptimizer
+
+import (
+	"go.uber.org/zap"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type OptimRegex struct {
+	Mainstats    *regexp.Regexp
+	GetCharNames *regexp.Regexp
+	Substats     *regexp.Regexp
+	Options      *regexp.Regexp
+}
+
+func InitRegex() *OptimRegex {
+	re := OptimRegex{}
+
+	// Regex to identify main stats based on flower. Check that characters all have one that we can recognize
+	re.Mainstats = regexp.MustCompile(`(?m)^[a-z]+\s+add\s+stats\s+hp=(4780|3571)\b[^;]*;`)
+	re.GetCharNames = regexp.MustCompile(`(?m)^([a-z]+)\s+char\b[^;]*;`)
+
+	// Regex to remove stat rows that do not look like mainstat rows from the config
+	re.Substats = regexp.MustCompile(`(?m)^[a-z]+\s+add\s+stats\b[^;]*;.*\n`)
+
+	re.Options = regexp.MustCompile(`([a-z_]+)=([0-9.]+)`)
+
+	return &re
+}
+
+func (re *OptimRegex) scrubSimCfg(cfg string, sugarLog *zap.SugaredLogger) string {
+	if len(re.Mainstats.FindAllString(cfg, -1)) != len(re.GetCharNames.FindAllString(cfg, -1)) {
+		sugarLog.Error("Error: Could not identify valid main artifact stat rows for all characters based on flower HP values.")
+		sugarLog.Error("5* flowers must have 4780 HP, and 4* flowers must have 3571 HP.")
+		os.Exit(1)
```

Comment:
I have to say.. the original usage of os.Exist(1) is not very Go like lol. Should really be returning an error in the function and handling that somehow (including exiting)